### PR TITLE
fixed bug of method PyString_FromString

### PR DESCRIPTION
**Pull request by yagweb**
### What does this implement/fix? Explain your changes.
 The second parameter of PyString_FromStringAndSize should be filled with the length of the bytes array, not the the length of the raw string.

This method is a very basic method，sufficient tests should be applied.

### Does this close any currently open issues?
#666 
A unit test method need be added.

### Any other comments?
No.

### Checklist

Check all those that are applicable and complete.

-   [ ] Make sure to include one or more tests for your change
-   [ ] If an enhancement PR, please create docs and at best an example
-   [ ] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [ ] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by denfromufa** [comment]
@yagweb failing in python 2.7:

https://travis-ci.org/pythonnet/pythonnet/jobs/377211034

`runtime.cs(1213,56): error CS1503: Argument 1: cannot convert from 'System.IntPtr' to 'string'`

**Comment by denfromufa** [comment]
@yagweb need a fix for appveyor pip update:

https://github.com/Microsoft/vscode-python/issues/1107

**Comment by denfromufa** [comment]
@yagweb will this PR fix these 3 disabled tests?

https://github.com/pythonnet/pythonnet/blob/6e4cf9d0fc4443db6f63c2f3c55e838fbda35ee0/src/embed_tests/TestPyString.cs#L38

https://github.com/pythonnet/pythonnet/blob/6e4cf9d0fc4443db6f63c2f3c55e838fbda35ee0/src/embed_tests/TestPyString.cs#L73

https://github.com/pythonnet/pythonnet/blob/6e4cf9d0fc4443db6f63c2f3c55e838fbda35ee0/src/embed_tests/TestPyAnsiString.cs#L89

**Comment by yagweb** [comment]
@denfromufa No.

I also found another bug of PyString/PyAnsiString, their ToString method don't work with non-ASCII strings.

I will add all these into the checklist and dig deeper when I have time. 

Maybe they are the cause of #662.

**Comment by denfromufa** [comment]
@yagweb i added the unit test based on reported bug by @EcmaXp in #666 but the issue does not look resolved in Python 2.7, works on all Python 3+ versions:

Travis CI:
`UnicodeDecodeError: 'ascii' codec can't decode byte 0xec in position 0: ordinal not in range(128)`
https://travis-ci.org/pythonnet/pythonnet/jobs/387536002#L2111

Appveyor CI:

```
assert test_unicode_str == text_type(world)
\x1b[1m\x1b[31mE       AssertionError: assert '\uc548\ub155' == '??'\x1b[0m
\x1b[1m\x1b[31mE         - \uc548\ub155\x1b[0m
E         + ??
src\tests\test_conversion.py:544: AssertionError
```
https://ci.appveyor.com/project/pythonnet/pythonnet/build/master-1182/job/nvwn4ek3eskteg3m#L220

What do you think? Maybe the test should be modified for Python 2.7?

**Comment by filmor** [comment]
I don't think this can work on Py2 without breaking backwards compatibility, as we identify `System.String` with `str` while it would need to be `unicode`.

**Comment by denfromufa** [comment]
@filmor agreed, so then skip the last test on PY2?

File: src/runtime/runtime.cs [hunk 1]
```diff
@@ -1193,7 +1193,11 @@ internal static bool PyString_Check(IntPtr ob)
 
         internal static IntPtr PyString_FromString(string value)
         {
+#if PYTHON3
+            return PyUnicode_FromKindAndData(_UCS, value, value.Length);
+#elif PYTHON2
             return PyString_FromStringAndSize(value, value.Length);
```
Comment (filmor) [inline src/runtime/runtime.cs]: Maybe check here whether the string is ASCII? I'm not sure where this one is used, but we should probably distinguish the case where we actually want Unicode (and should also return that on Python 2) from the cases that should be ASCII-only on Python 2 (like identifiers).

Comment (denfromufa) [reply src/runtime/runtime.cs]: @filmor can you file a separate issue for this with a suggested fix? this PR is now ready to merge!

**Pull request merged**